webui/react/src/components/ActionDropdown.tsx

```diff
@@ -0,0 +1,93 @@
+import { Dropdown, Menu, Modal, ModalFuncProps } from 'antd';
+import { MenuInfo } from 'rc-menu/lib/interface';
+import React, { JSXElementConstructor } from 'react';
+
+import Icon from 'components/Icon';
+import handleError, { ErrorLevel, ErrorType } from 'ErrorHandler';
+import { capitalize } from 'utils/string';
+import { Eventually } from 'utils/types';
+
+import css from './ActionDropdown.module.scss';
+
+// TODO parameterize Action using Enums? https://github.com/microsoft/TypeScript/issues/30611
+export type Triggers<T extends string> = Partial<{ [key in T]: () => Eventually<void> }>
+export type Confirmations<T extends string> =
+  Partial<{ [key in T]: Omit<ModalFuncProps, 'onOk'>}>
+
+interface Props<T extends string> {
+  actionOrder: T[];
+  confirmations?: Confirmations<T>
+  onComplete?: (action?: T) => void;
+  onTrigger: Triggers<T>;
+  thing: { id: string, kind: string };
+}
+
+const stopPropagation = (e: React.MouseEvent): void => e.stopPropagation();
+
+const ActionDropdown = <T extends string>(
+  { thing: { id, kind }, onComplete, onTrigger, confirmations, actionOrder }: Props<T>,
+): React.ReactElement<unknown, JSXElementConstructor<unknown>> | null => {
+
+  const handleMenuClick = async (params: MenuInfo): Promise<void> => {
+    params.domEvent.stopPropagation();
+    try {
+      const action = params.key as T;
+      const fn = onTrigger[action];
```

Comment:
**super nit:** `trigger` or `handleTrigger` instead of `fn`?

Reply:
done. (in the other pr) https://github.com/determined-ai/determined/pull/3194